### PR TITLE
feat(k3d): Support registry-content from UI

### PR DESCRIPTION
**Pull request by orangedeng**
Related issue: https://github.com/cnrancher/autok3s/issues/577
And minor fixes for k3d provider codes.

### CODE REVIEW

**Comment by orangedeng** [comment]
Already tested cli and UI function and it works.

File: pkg/utils/registry.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+func VerifyRegistryFileContent(path, content string) (*k3s.Registry, error) {
+	var registry k3s.Registry
+	var err error
+	contentBytes := []byte(content)
+	if path != "" && content == "" {
+		contentBytes, err = ioutil.ReadFile(path)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return &registry, nil
+			}
```
Comment (Jason-ZW) [inline pkg/utils/registry.go:21]: 这块为什么不反回 err？

**Review by Jason-ZW — CHANGES_REQUESTED** [review]
历史数据兼容性肯能需要测试下，Others LGTM

File: pkg/utils/registry.go (hunk 1 above)
Comment (orangedeng) [reply pkg/utils/registry.go:21]: 这个处理是复用原方法，没有做改动

**Review by Jason-ZW — APPROVED** [review]

**Pull request merged**